modules/core/src/main/java/org/apache/synapse/mediators/AbstractListMediator.java

```diff
@@ -109,6 +117,14 @@ public boolean mediate(MessageContext synCtx, int mediatorPosition) {
                     synCtx.setTracingState(myEffectiveTraceState);
                     if (!mediator.mediate(synCtx)) {
                         returnVal = false;
+                        if (i == mediators.size() - 1) {
```

Comment:
are we notifying the completeness if the last mediator returns a true?

Reply:
This is fixed